src/main/java/dinkplugin/notifiers/SpeedrunNotifier.java

```diff
@@ -48,27 +52,38 @@ public void onWidgetLoaded(WidgetLoaded event) {
     }
 
     private void attemptNotify(String questName, String duration, String pb) {
-        Duration bestTime = TimeUtils.parseTime(pb);
-        Duration currentTime = TimeUtils.parseTime(duration);
-        boolean isPb = bestTime.compareTo(currentTime) >= 0;
-        if (!isPb && config.speedrunPBOnly()) {
+        if (!isPersonalBest && config.speedrunPBOnly()) {
             return;
         }
 
         // pb or notifying on non-pb; take the right string and format placeholders
         Template notifyMessage = Template.builder()
-            .template(isPb ? config.speedrunPBMessage() : config.speedrunMessage())
+            .template(isPersonalBest ? config.speedrunPBMessage() : config.speedrunMessage())
             .replacementBoundary("%")
             .replacement("%USERNAME%", Replacements.ofText(Utils.getPlayerName(client)))
             .replacement("%QUEST%", Replacements.ofWiki(questName))
             .replacement("%TIME%", Replacements.ofText(duration))
             .replacement("%BEST%", Replacements.ofText(pb))
             .build();
 
+        // Reformat the durations for the extra object
+        Duration bestTime = TimeUtils.parseTime(pb);
+        Duration currentTime = TimeUtils.parseTime(duration);
+
         createMessage(config.speedrunSendImage(), NotificationBody.builder()
             .text(notifyMessage)
-            .extra(new SpeedrunNotificationData(questName, bestTime.toString(), currentTime.toString()))
+            .extra(new SpeedrunNotificationData(questName, bestTime.toString(), currentTime.toString(), isPersonalBest))
             .type(NotificationType.SPEEDRUN)
             .build());
     }
+
+    public void onGameMessage(String chatMessage) {
+        if (!isEnabled()) {
+            return;
+        }
+
+        if (chatMessage.startsWith("Speedrun duration: ")) {
+            isPersonalBest = chatMessage.endsWith(" (new personal best)");
```

Comment:
:+1: 

If game message could occur after widget load, we may want `isPersonalBest` to be initialized as null (but this seems unlikely)

Reply:
I think this makes sense, working on this now

Reply:
Not implementing this since it'd increase the complexity of a simple notifier - instead we'll just keep an eye out on false isPersonalBest values & fix it if it comes up